include/deal.II/lac/vector_space_vector.h

```diff
@@ -0,0 +1,44 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_vector_space_vector_h
+#define dealii_vector_space_vector_h
+
+#include <deal.II/base/config.h>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+// ---------------------------- Free functions --------------------------
+
+namespace LinearAlgebra
+{
+  /**
+   * Shift all entries of the vector by a constant factor so that the mean
+   * value of the vector becomes zero.
+   */
+  template <typename VectorType>
+  void
+  set_zero_mean_value(VectorType &vector)
```

Comment:
Did we agree on deprecating this feature, as suggested by https://github.com/dealii/dealii/pull/15688#discussion_r1258201969?
```suggestion
  template <typename VectorType>
  DEAL_II_DEPRECATED void
  set_zero_mean_value(VectorType &vector)
```

Reply:
Actually, while I dislike the name of the function, I think `lac/` is the right place for it and `VectorTools` is not. The latter deals with finite element fields, which are defined on a mesh and with regard to a `DoFHandler`. This function is defined on a purely algebraic vector level. I think this is the right directory for it, if perhaps not the right file.

Reply:
@kronbichler @bangerth *ping*

Reply:
I think `DEPRECATED_EARLY` is our standard now. We also need to print a warning when including this file.